File: Codenotch.Maui/Models/NotchVisibility.cs
namespace Codenotch.Models;

/// <summary>
/// When the notch appears on screen.
/// Converted from Settings/NotchVisibility.swift.
/// </summary>
public enum NotchVisibility
{
    AlwaysShow,
    OnHover,
    Hidden
}

public static class NotchVisibilityExtensions
{
    public static string Title(this NotchVisibility visibility) => visibility switch
    {
        NotchVisibility.AlwaysShow => "Always",
        NotchVisibility.OnHover => "On hover",
        NotchVisibility.Hidden => "Hidden",
        _ => visibility.ToString()
    };

    public static string Explanation(this NotchVisibility visibility) => visibility switch
    {
        NotchVisibility.AlwaysShow => "Stays on screen, readings visible all the time.",
        NotchVisibility.OnHover => "Folds into a narrow black pill at the edge until you move the pointer over it.",
        NotchVisibility.Hidden => "Removes the notch entirely.",
        _ => string.Empty
    };
}

File: lib/settings/notch_visibility.dart
enum NotchVisibility {
  onHover('On hover', 'Folds into a small pill at rest; unfolds when the pointer reaches it.'),
  alwaysShow('Always show', 'Stays open at full size, showing all provider rings.'),
  hidden('Hide', 'Keeps the notch off screen entirely. Reopening the app opens settings.');

  final String title;
  final String explanation;

  const NotchVisibility(this.title, this.explanation);
}
